src/libmerklecpp/merklecpp.h

```diff
@@ -0,0 +1,1294 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+
+#pragma once
+
+#include <array>
+#include <cassert>
+#include <cmath>
+#include <cstddef>
+#include <cstdint>
+#include <cstring>
+#include <functional>
+#include <list>
+#include <memory>
+#include <sstream>
+#include <stack>
+#include <vector>
+
+#ifdef WIN32
+#  include <stdlib.h>
+#  if BYTE_ORDER == LITTLE_ENDIAN
+#    define htobe32(X) _byteswap_ulong(X)
+#    define be32toh(X) _byteswap_ulong(X)
+#  else
+#    define htobe32(X) (X)
+#    define be32toh(X) (X)
+#  endif
+#  undef max
+#endif
+
+#ifdef WITH_TRACE
+#  include <iostream>
+#  define TRACE_HASH_SIZE 3
+#  define TRACE(X) \
+    { \
+      X; \
```

Comment:
It'd be better if this way defined to use the logging macros (LOG_TRACE_FMT here), since they know to use the ringbuffer when inside the enclave and there is no std::out. That perhaps belong in a separate file that's effectively CCF integration if we want to keep this independent/spin it off at some point, but I think it would still be a good thing.